src/iinit.cpp

```diff
@@ -66,11 +69,226 @@ mkrodsdir() {
-    printf( "missing; please enter them.\n" );
-}
+namespace
+{
+    auto save_updates_to_irods_environment(const nlohmann::json& _update) -> void
+    {
+        std::string env_file;
+        std::string session_file;
+        if (auto ret = irods::get_json_environment_file(env_file, session_file); !ret.ok()) {
+            fmt::print("failed to get environment file - [{}]\n", ret.code());
+        }
+
+        json obj_to_dump;
+
+        if (std::ifstream in{env_file}; in) {
+            try {
+                in >> obj_to_dump;
+            }
+            catch (const json::parse_error& e) {
+                obj_to_dump = _update;
+                fmt::print(
+                    stderr,
+                    "Failed to parse environment file: []\n"
+                    "Falling back to original environment settings.",
+                    e.what());
+            }
+
+            obj_to_dump.merge_patch(_update);
+        }
+        else {
+            obj_to_dump = _update;
+        }
+
+        if (std::ofstream f(env_file.c_str(), std::ios::out); f.is_open()) {
+            f << obj_to_dump.dump(4) << "\n";
+            f.close();
+        }
+        else {
+            fmt::print("Failed to save environment file [{}]\n", env_file);
+        }
+    } // save_updates_to_irods_environment
+
+    auto option_specified(std::string_view _option, int argc, char** argv) -> bool
+    {
+        for (int arg = 0; arg < argc; ++arg) {
+            if (!argv[arg]) {
+                continue;
+            }
+
+            if (_option == argv[arg]) {
+                // parseCmdLineOpt is EVIL and requires this. Please don't ask why.
+                argv[arg] = "-Z";
+                return true;
+            }
+        }
+
+        return false;
+    } // option_specified
+
+    auto set_env_from_prompt(char _setting[], const char* _prompt, std::size_t _len) -> void
+    {
+        // If the setting already has a value, use that as the default value and show it in the prompt.
+        if (0 == std::strlen(_setting)) {
+            fmt::print("{}: ", _prompt);
+        }
+        else {
+            fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), _setting);
+        }
+
+        std::string response;
+        std::getline(std::cin, response);
+
+        // If the response is empty, don't copy anything. If there is a value in _setting already, it will be used
+        // as the default response. If there is nothing, then there is nothing to copy anyway.
+        if (!response.empty()) {
+            std::strncpy(_setting, response.c_str(), _len);
+        }
+    } // set_env_from_prompt
+
+    auto set_env_from_prompt(char _setting[], const char* _default, const char* _prompt, std::size_t _len) -> void
+    {
+        const bool env_has_value = 0 != std::strlen(_setting);
+        const char* default_value = env_has_value ? _setting : _default;
+
+        fmt::print(fmt::runtime(fmt::format("{} [{{}}]: ", _prompt)), default_value);
```

Comment:
Seems this line could be simplified?
```cpp
fmt::print("{} [{}]: ", _prompt, default_value);
```

Reply:
Hmm... yes, I think you're right. This is probably a holdover from a previous iteration of this logic. Will update.